src/transform/src/nonnull_requirements.rs

```diff
@@ -195,16 +195,58 @@ impl NonNullRequirements {
                 expected_group_size: _,
             } => {
                 let mut new_columns = HashSet::new();
-                for column in columns {
-                    // No obvious requirements on aggregate columns.
-                    // A "non-empty" requirement, I guess?
-                    if column < group_key.len() {
-                        group_key[column].non_null_requirements(&mut new_columns);
+                let (group_key_columns, aggr_columns): (Vec<usize>, Vec<usize>) =
+                    columns.iter().partition(|c| **c < group_key.len());
+                for column in group_key_columns {
+                    group_key[column].non_null_requirements(&mut new_columns);
+                }
+
+                if !aggr_columns.is_empty() {
+                    // Intersect the non-null requirements for non-null aggregates
+                    // (one which its result is requested to be non-null)
+                    // that propagate non-null constraints (all of them but COUNT).
+                    // However, if there is a COUNT or any other aggregate we can still
+                    // propagate the non-null requirement as long as the requirements
+                    // of their parameters intersect with the requirements of a non-null
+                    // aggregate.
+                    let aggr_nonnull_req = aggregates
+                        .iter()
+                        .enumerate()
+                        .map(|(pos, aggr)| {
+                            let aggr_nonnull_flag = aggr.func.propagates_nonnull_constraint()
+                                && aggr_columns.contains(&(group_key.len() + pos));
+                            let mut aggr_nonnull_columns = HashSet::new();
+                            aggr.expr.non_null_requirements(&mut aggr_nonnull_columns);
+                            (aggr_nonnull_columns, aggr_nonnull_flag)
+                        })
+                        .collect::<Vec<_>>();
+                    let mut req_intersection: Option<HashSet<usize>> = None;
```

Comment:
Minor nit that `req_intersection` as a name describes what it is happening to the thing, rather than what the thing represents. I think what it means to represent is "candidate input columns whose nullness would allow dropping a row".

Reply:
I've renamed `req_intersection` as `pushable_nonnull_constraints`. Comment reads as follows

```
                        // If no aggregate prevents other non-null pushdowns, `pushable_nonnull_constraints`
                        // contains the columns whose nulls are ignored by all aggregates that can push down
                        // non-null constraints.

```